src/dataset.js

```diff
@@ -221,6 +221,31 @@ function Dataset(bigQuery, id) {
 
 util.inherits(Dataset, common.ServiceObject);
 
+/**
+ * Start running a query scoped to your dataset.
```

Comment:
We should copy the gist of the parent method's description:

 * Run a query as a job. No results are immediately returned. Instead, your
 * callback will be executed with a {@link Job} object that you must
 * ping for the results. See the Job documentation for explanations of how to
 * check on the status of the job.

Then link to the upstream.

Reply:
Done - PTAL!